include/seastar/core/chunked_fifo.hh

```diff
@@ -146,6 +146,7 @@ private:
         size_t _item_index = 0;
 
     protected:
+        inline basic_iterator() noexcept;
```

Comment:
the reason we need to have an uninitialized iterator is that we need to create an empty `context` at https://github.com/scylladb/seastar/blob/e908cfce2d4488976f3b91906954f435aaa99238/src/core/scollectd.cc#L418 , and the `context` has an `seastar::metrics::impl::metric_metadata_fifo::iterator` after this change. without a default ctor, `make_lw_shared<context>()` won't compile.

Reply:
i concur with @nyh that it'd be better to explain why we need a default ctor for the `basic_iterator`. as it might be otherwise not self-explained.

Reply:
Yes, I guessed that some other code must have used an uninitialized iterator and initialize it later, and since this works with std::vector (and I didn't check, probably also with std::dequeue) then it should also work with chunked_fifo, so it's a good change.

Reply:
Yes, I added the default ctor to the iterator because of compilation issues with `make_lw_shared<context>()`. I will split up the commits and explain that further in the commit message.